### PR TITLE
ARXIVNG-1359 store submission content in the right location

### PR DESCRIPTION
**Pull request by erickpeirson**
This introduces a simple function for storing submission content in the right spot to make the legacy system happy. The idea is that the submission UI application (deployed on classic infrastructure) will use this when the submission is finalized. Something like (probably in an async worker):
- Retrieve source package and log from file management service
- Retrieve PDF from compilation service
- Use this function to store things on the filesystem

The composition will need to be implemented in the submission UI project.

### CODE REVIEW

**Comment by erickpeirson** [comment]
@martin @DavidLFielding @bdc34 Would be great to get your eyes on this to see if I'm interpreting the situation correctly. Also, there are probably more failure conditions to check for, so please fire away.

File: core/arxiv/submission/services/filesystem/store.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                 log_path: str) -> None:
+    """
+    Deposit a source package into the legacy filesystem.
+
+    Parameters
+    ----------
+    submission_id : int
+        Numeric submission ID. This is **not** the upload ID.
+    source_path : str
+        Path to a gzipped tarball containing the source package to deposit.
```
Comment (mhl10) [inline core/arxiv/submission/services/filesystem/store.py]: This implies that only tarballed source packages are supported. How about pdf-only or html submissions?

File: core/arxiv/submission/services/filesystem/store.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    shard_dir = os.path.join(base_dir, str(submission_id)[:4])
+    source_dir = os.path.join(shard_dir, str(submission_id))
+
+    # Make sure that we have a place to put the source files.
+    if not os.path.exists(source_dir):
+        os.makedirs(source_dir)  # Recursively create directories.
+
+    # Deposit the source package. The classic system expects a directory called
+    # "src".
+    tf = tarfile.open(source_path, 'r:gz')
```
Comment (mhl10) [inline core/arxiv/submission/services/filesystem/store.py]: Per my previous comment, probably need to handle the pdf-only and html cases here. For pdf-only, it's just a matter of copying the source pdf to `f'{submission_id}'.pdf` in `source_dir`. I'll need to check what happens with html submissions.

Comment (erickpeirson) [reply core/arxiv/submission/services/filesystem/store.py]: Ah, indeed; thanks! The assumption here is that we're getting a source package from the file management service. Even for a PDF-only submission, its content download endpoint should return a gzipped tarball. But I overlooked the fact that it doesn't go in ``source_dir/src``.

**Comment by erickpeirson** [comment]
@mhl10 I think that this should handle both the conventional and PDF-only scenarios.

For PDF-only submissions, we'll be grabbing the PDF and depositing it to the preview service. So the end result here should be that:
- the PDF ends up at ``/1234/1234567/1234567.pdf``
- the source is unpacked into ``src``, so the original PDF ends up at ``/1234/1234567/src/original_filename.pdf``

This looks right to me when I inspect the current production filesystem. But I admit that I did not go back to the legacy code this time, so I may have misinterpreted.

File: agent/agent/services/filesystem/filesystem.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        Verifies the integrity of the transferred file by comparing the content
+        of the ``ETag`` response  header to ``checksum``.
+        """
+        response = self.request('post', f'/{submission_id}/preview',
+                                data=pointer)
+        etag = response.headers.get('ETag')
+        if etag != checksum:
+            raise ValidationFailed(f'Expected {checksum}, got {etag}')
+
+    def source_exists(self, submission_id: int,
```
Comment (mhl10) [inline agent/agent/services/filesystem/filesystem.py]:
```suggestion
    def does_source_exist(self, submission_id: int,
```
(or `has_source`)

File: agent/agent/services/filesystem/filesystem.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        ``checksum``.
+        """
+        response = self.request('head', f'/{submission_id}/source')
+        if checksum is not None:
+            etag = response.headers.get('ETag')
+            if etag != checksum:
+                raise ValidationFailed(f'Expected {checksum}, got {etag}')
+        return bool(response.status_code == status.OK)
+
+    def preview_exists(self, submission_id: int,
```
Comment (mhl10) [inline agent/agent/services/filesystem/filesystem.py]:
```suggestion
    def does_preview_exist(self, submission_id: int,
```
(or `has_preview`)

File: filesystem/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Legacy filesystem integration shim
+
+The goal of this app is to make sure that the NG submission system can put
+files in the right place on the legacy filesystem when a submission is
```
Comment (mhl10) [inline filesystem/README.md]: perhaps more emphatically:
```suggestion
files in the right place on the legacy filesystem when (and only when) a submission is
```

File: filesystem/filesystem/config.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+import os
+import tempfile
+
+LEGACY_FILESYSTEM_ROOT = os.environ.get('LEGACY_FILESYSTEM_ROOT')
+if LEGACY_FILESYSTEM_ROOT is None:
+    LEGACY_FILESYSTEM_ROOT = tempfile.mkdtemp()
+
+LEGACY_FILESYSTEM_SOURCE_DIR_MODE = \
+    int(os.environ.get('LEGACY_FILESYSTEM_SOURCE_DIR_MODE', '17917'))
```
Comment (mhl10) [inline filesystem/filesystem/config.py]: What does mode `17917` translate to?

Comment (mhl10) [reply filesystem/filesystem/config.py]: In any case, it may be more legible as an octal string.

Comment (erickpeirson) [reply filesystem/filesystem/config.py]: ``0o42775``. Fixed in 8ec0e04

File: agent/agent/services/filesystem/filesystem.py (hunk 4 above)
Comment (erickpeirson) [reply agent/agent/services/filesystem/filesystem.py]: fixed in 161a886

File: agent/agent/services/filesystem/filesystem.py (hunk 3 above)
Comment (erickpeirson) [reply agent/agent/services/filesystem/filesystem.py]: fixed in 161a886

**Pull request merged**